### PR TITLE
Replaced babel presets with preset-env 

### PR DESCRIPTION
**Pull request by awolkers**
- Replaced deprecated presets with the new preferred preset-env.
- Added target browsers (browserlist format) to .babelrc 
- Updated some related babel packages.
- Deleted unused babel packages.
- .browserslistrc tweak: added `not dead` query (browsers from last `2 version` query, but with less than 0.5% in global usage statistics and without official support or updates for 24 months.)

### CODE REVIEW

**Comment by awolkers** [comment]
slack hook test

File: package.json [hunk 1]
```diff
@@ -36,11 +36,11 @@
     "autoprefixer": "^8.3.0",
     "babel-core": "^6.26.3",
     "babel-eslint": "^8.2.3",
-    "babel-plugin-external-helpers": "^6.8.0",
-    "babel-polyfill": "^6.16.0",
-    "babel-preset-es2015": "^6.24.0",
-    "babel-preset-es2016": "^6.22.0",
-    "babel-preset-stage-2": "^6.5.0",
+    "babel-preset-env": "^1.7.0",
+    "babel-preset-stage-3": "^6.24.1",
```
Comment (wolfpilot) [inline package.json]: As none of the **stage-x** presets are configured in `.babelrc`, this dependency should either be removed or the stage added to the config.

Personally, I would recommend removing it entirely. **stage-x** presets have always been a maintenance concern, it's also the reason why they're being  [dropped](https://babeljs.io/blog/2018/07/27/removing-babels-stage-presets) starting with Babel v7.

Comment (awolkers) [reply package.json]: The `babel-preset-stage-3` is needed so we can use the `transform-object-rest-spread` plugin within the `.babelrc`.

Comment (wolfpilot) [reply package.json]: Indeed, my bad. It's currently in [stage 4](https://github.com/tc39/proposal-object-rest-spread), we should probably update this as soon as it's implemented in the new revision. Thanks for looking into this 👍

**Comment by colin-aarts** [comment]
Why is the browserslist now duplicated in another location?

**Comment by awolkers** [comment]
@colin-aarts Babel 6 doesn't support an external config (`.browserslistrc` or `package.json) when it comes to browserslist. When we upgrade to Babel 7 in theory we could get rid of the config within the `.babelrc` file. However there might be cases where you want the css and js support to be different. For instance javascript is supported from > IE11 but we still want to use the flexbox autoprefixer for IE10. For now we should leave this as is.

File: package.json (hunk 1 above)
Comment (peeke) [reply package.json]: Just tested this, but it's not needed. You just need to add the plugin separately
You should add `babel-plugin-transform-object-rest-spread`, then you can remove `babel-preset-stage-3`.

**Comment by awolkers** [comment]
@peeke replaced package babel-preset-stage-3 with babel-plugin-transform-object-rest-spread

**Pull request merged**